test/powershell/Language/Scripting/CheckRestrictedlanguage.Tests.ps1

```diff
@@ -0,0 +1,112 @@
+﻿#  <Test>
+#    <summary>Test restricted language check method on scriptblocks</summary>
+#  </Test>
+Describe "Test restricted language check method on scriptblocks" -Tags "CI" {
+        BeforeAll {
+            set-strictmode -v 2
+            function list {
+
+            $l = new-object system.collections.generic.list[string]
+            $args | foreach {$l.Add($_)}
+            , $l
+            }
+        }
+        
+
+        It 'Check basic expressions' {
+
+            {{2+2}.CheckRestrictedLanguage($null, $null, $false) } | Should Not Throw  # Succeed with no variables
+        }
+        
+        It 'Check default variables' {
+
+            {{ $PSCulture, $PSUICulture, $true, $false, $null}.CheckRestrictedLanguage($null, $null, $false) } | Should Not Throw
+        }
+
+        It 'Check default variables' {
+            
+            try
+            {
+                {2+$a}.CheckRestrictedLanguage($null, $null, $false)
+                Throw "Execution OK"
+            }
+            catch
+            {
+                $_.FullyQualifiedErrorId | Should Be 'ParseException'
+            }
+            
+        }
+
+        It 'Check union of default + one allowed variables' {
+
+            { { 2 + $a }.CheckRestrictedLanguage($null, (list a), $false) }| Should Not Throw  # succeed
+        }
+
+        It 'Check union of default + two allowed variables' {
+
+            { { $a + $b }.CheckRestrictedLanguage($null, (list a b), $false) } | Should Not Throw  # succeed
+        }
+
+        It 'Check union of default + allowed variables' {
+
+            { { $PSCulture, $PSUICulture, $true, $false, $null,  $a, $b}.CheckRestrictedLanguage($null, (list a b), $false) }| Should Not Throw
+        }
+
+        It 'Check union of default + one disallowed variables' {
+
+            try
+            {
+                { $a + $b + $c }.CheckRestrictedLanguage($null, (list a b), $false)   # fail
+                Throw "Execution OK"
+            }
+            catch
+            {
+                $_.FullyQualifiedErrorId | Should Be 'ParseException'
+            }
+        }
+
+        It 'Check union of default + one allowed variable and but not allow evironment variable' {
```

Comment:
fixed

---
In reply to: [76495139](https://github.com/powershell/powershell/pull/2086#discussion_r76495139) [](ancestors = 76495139)